jwql/instrument_monitors/common_monitors/bias_monitor.py

```diff
@@ -0,0 +1,554 @@
+            # in the calibrated image
+            cal_data = fits.getdata(processed_file, 'SCI')[0, 0, :, :]
+            dq = fits.getdata(processed_file, 'PIXELDQ')
+            mean, median, stddev = sigma_clipped_stats(cal_data[dq==0], sigma=3.0, maxiters=5)
+            logging.info('\tCalculated calibrated image stats: {:.3f} +/- {:.3f}'
+                         .format(mean, stddev))
+            collapsed_rows, collapsed_columns = self.collapse_image(cal_data)
+            logging.info('\tCalculated collapsed row/column values of calibrated image.')
+
+            # Save a png of the calibrated image for visual inspection
+            logging.info('\tCreating png of calibrated image')
+            output_png = self.image_to_png(cal_data, 
+                outname=os.path.basename(processed_file).replace('.fits',''))
+
+            # Construct new entry for this file for the bias database table.
+            # Can't insert values with numpy.float32 datatypes into database 
+            # so need to change the datatypes of these values.
+            bias_db_entry = {'aperture': self.aperture,
+                             'uncal_filename': filename,
+                             'cal_filename': processed_file,
+                             'cal_image': output_png,
+                             'expstart': expstart,
+                             'mean': float(mean),
+                             'median': float(median),
+                             'stddev': float(stddev),
+                             'collapsed_rows': collapsed_rows.astype(float),
+                             'collapsed_columns': collapsed_columns.astype(float),
+                             'entry_date': datetime.datetime.now()
+                            }
+            for key in amp_meds.keys():
+                bias_db_entry[key] = float(amp_meds[key])
+            
+            # Add this new entry to the bias database table
+            self.stats_table.__table__.insert().execute(bias_db_entry)
+            logging.info('\tNew entry added to bias database table: {}'
+                         .format(bias_db_entry))
+
+    @log_fail
+    @log_info
+    def run(self):
+        """The main method.  See module docstrings for further details.
+        """
+
+        logging.info('Begin logging for bias_monitor')
+
+        # Get the output directory and setup a directory to store the data
+        self.output_dir = os.path.join(get_config()['outputs'], 'bias_monitor')
+        ensure_dir_exists(os.path.join(self.output_dir, 'data'))
+
+        # Use the current time as the end time for MAST query
+        self.query_end = Time.now().mjd
+
+        # Loop over all instruments
+        for instrument in ['nircam']:
+            self.instrument = instrument
+
+            # Identify which database tables to use
+            self.identify_tables()
+
+            # Get a list of all possible full-frame apertures for this instrument
+            s = Siaf(self.instrument)
```

Comment:
It is frowned upon to use single letter variable names.  Could you use something more descriptive here?  Perhaps `siaf`? or `aperture_list`?

Reply:
ah, good to know - switched to siaf